docs/operators/windowedOperator.md

```diff
@@ -0,0 +1,140 @@
+
+### `SessionWindows`
+
+`SessionWindow`s have variable durations and are based on the key of the tuple. Each tuple is assigned to exactly one window. It takes a duration parameter `minGap`, which specifies the minimum time gap between two `SessionWindow`s of the same key. To ensure that no two `SessionWindow`s of the same key are less than `minGap` apart, upon an arrival of a tuple, the `WindowedOperator` does the following checks:
+  
+#### The arrival of the new tuple would result in a merge of the two existing session windows
+
+![](images/windowedOperator/session-windows-1.png) 
+
+A new Session Window is created with the merged state of the two existing `SessionWindow`s of the same key, plus the new tuple. The two existing `SessionWindow`s will be deleted and retraction triggers for the two deleted windows will be fired. (Please see [here](#triggers) for details on `Trigger`s)
+
+#### The arrival of the new tuple would result in an extension of an existing `SessionWindow`s
+
+![](images/windowedOperator/session-windows-2.png) 
+
+A new `SessionWindow` is created with the state of the existing `SessionWindow`, plus the new tuple, with a longer duration than the existing `SessionWindow` to cover the new tuple. The existing `SessionWindow` will be deleted and a retraction trigger for the old window will be fired.
+
+#### The new tuple can safely be assigned to an existing `SessionWindow` without change
+
+![](images/windowedOperator/session-windows-3.png) 
+
+The new tuple is simply applied to the state of the existing `SessionWindow`.
+
+If all of the above checks return false, then a new `SessionWindow` is created for the new tuple.
+
+## Timestamp Extractor
+
+The `WindowedOperator` expects a timestamp extractor. This is for `WindowedOperator` to extract the timestamp from the tuple for window assignment.
+
+## Watermarks
+
+Watermarks are control tuples that includes a timestamp. It basically tells `WindowedOperator` that all windows that lie completely before the given timestamp are considered late, and the rest of the windows are considered early. 
+
+### Fixed Watermark
+
+If watermarks are not available from upstream, the user of the WindowedOperator can set a fixed watermark. The fixed watermark represents the number of milliseconds before the timestamp derived from the Apex streaming window ID. Note that the Apex streaming window ID is an implicit timestamp that more or less represents the ingression time of the tuple.
+
+## Allowed Lateness
+
+![](images/windowedOperator/allowed-lateness.png) 
+
+Allowed Lateness specifies the lateness horizon from the watermark. If a tuple has a timestamp that lies beyond the lateness horizon, it is dropped by the `WindowedOperator`. Also, if a window completely lies beyond the lateness horizon, the window along with its state is purged from `WindowedOperator`.
```

Comment:
For a window to be completely beyond the lateness horizon, how would it become part of the state?